pkg/jobworker/tail.go

```diff
@@ -0,0 +1,40 @@
+package jobworker
+
+import (
+	"io"
+	"os"
+	"time"
+)
+
+// tailReader wraps the normal io.ReadCloser with an implementation that sleeps for a specified pollInterval
+// before retrying to Read
+type tailReader struct {
+	io.ReadCloser
+	pollInterval time.Duration
+}
+
+// Read calls the normal io.ReadCloser and checks for an io.EOF error and skips returning, to sleep instead
+func (t tailReader) Read(b []byte) (int, error) {
+	for {
+		n, err := t.ReadCloser.Read(b)
+		if n > 0 {
+			return n, nil
+		} else if err != io.EOF {
+			return n, err
+		}
+		time.Sleep(t.pollInterval)
+	}
```

Comment:
If a job exits or has no more output will this block forever? Are there ways to implement this without polling?

Reply:
No, since we return an io.ReadCloser it's the callers responsibility to call Close() which will cause the Read function to return an error.

In my example at `cmd/example/main.go` I `defer reader.Close()` after calling `job.Output()`, the example catches Ctrl+C then stops the job and updates the wait group counter with Done. Once this happens the blocking call on main wg.Wait() returns, the deferred Close() will be called and therefore the next call to Scan in the logging go routine will be false causing the go routine to complete.

Reply:
The only other two implementations without adding dependencies I could think of were:

1) Use inotify to watch for file changes (which tail uses under the hood) instead of polling. This was rejected for complexity of implementation and adding dependency on inotify (FreeBSD uses kevents)

2) Actually perform `tail -F` using exec.Cmd and returning STDOUT. I thought this was a little hacky and not good for performance

Reply:
Agreed inotify/tail are probably more complex/hacky than its worth here. Thanks for the thought process though.

As for the reading blocking until the ReadCloser is closed, I think this does impact the API a bit where retrieving logs hangs until the client decides to end the stream instead of being able to close the stream server side with a good status code when the end of the log file is reached and there will be no more additional output.

Not a big deal if its to much work to fix this, but I am curious to hear your thoughts on how this could be improved.

Reply:
I'm not sure if adding an extra flag improves the UX much. The behavior I'd expect is like `docker logs -f`. If the command quits, the reader gets io.EOF. Otherwise, the reader waits for more output.

Without that UX, the use case for remote commend execution seems limited. E.g., I can't reliably start a command and capture its output to a file without waiting for it to quit somehow.

Reply:
OK I understand the difference. Please take a look at my latest update which should resolve this. I now keep a reference to the readers in Job so that when the command completes, the reader's are closed preventing them from blocking.

The example now exhibits the behaviour of docker where the reader will exit, due to the log file being closed, when the command completes, or is stopped